openedx/frontend-app-authoring
# feat: block creation in libraries v2

## Description

This PR changes the way components are creating in libraries V2 with the purpose to avoid the creation of a blank component as fist step, behavior that have the following problems:

- The "cancel" action doesn't cancel the addition of the component, which seems like a bug.
- We can't set a https://github.com/openedx/frontend-app-authoring/issues/1384 for the component, because its title is not set at the time of creation.

## Supporting information

This PR is liked to https://github.com/openedx/frontend-app-authoring/issues/1482#issuecomment-2512498837 issue. 

## Testing instructions

- Create a component in libraries V2 supported by the editor `[text, video, problem]` 
- Press the cancel button or the close icon, no new component should appear in the platform. 
- Prove the right creating workflow:
   - Update the title and save it, should be used as `usage_key` block.
   - If the title does not change should create a unique key.
   - The user is enable to update images.
   - Press the save button should create the component and save the content.

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.

### PR sandbox

UN/PW: openedx / openedx

**Settings**
```yaml
```
